cpu/stm32_common/stmclk_common.c

```diff
@@ -47,35 +47,35 @@ void stmclk_disable_hsi(void)
 {
     /* we only disable the HSI clock if not used as input for the PLL and if
      * not used directly as system clock */
-#if CLOCK_HSE
-    if ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSI) {
-        RCC->CR &= ~(RCC_CR_HSION);
+    if (CLOCK_HSE) {
```

Comment:
IIRC some compilers, e.g. clang, may complain about `tautological compare` if CLOCK_HSE is `0` rendering this `if` always false and hence obsolete.